solr-search/src/test/java/edu/unc/lib/dl/search/solr/service/MultiSelectFacetListServiceIT.java

```diff
@@ -0,0 +1,537 @@
+        assertFacetValueCount(resp, CONTENT_TYPE, "/text^pdf", 1);
+
+        assertNumberFacetsReturned(resp, ROLE_GROUP, 3);
+        assertFacetValueCount(resp, ROLE_GROUP, "canViewOriginals|everyone", 1);
+        assertFacetValueCount(resp, ROLE_GROUP, "unitOwner|unitOwner", 1);
+        assertFacetValueCount(resp, ROLE_GROUP, "canManage|manager", 1);
+    }
+
+    @Test
+    public void noMatchesParentCollectionTest() throws Exception {
+        SearchState searchState = new SearchState();
+        searchState.setFacetsToRetrieve(FACETS_TO_RETRIEVE);
+        searchState.setFacet(new GenericFacet(PARENT_COLLECTION.name(), "doesnotexist"));
+
+        SearchRequest request = new SearchRequest(searchState, accessGroups);
+        SearchResultResponse resp = service.getFacetListResult(request);
+
+        assertNumberFacetsReturned(resp, PARENT_COLLECTION, 2);
+        assertFacetValueCount(resp, PARENT_COLLECTION, testCorpus.coll1Pid.getId(), 6);
+        assertFacetValueCount(resp, PARENT_COLLECTION, testCorpus.coll2Pid.getId(), 5);
+
+        assertNumberFacetsReturned(resp, CONTENT_TYPE, 0);
+
+        assertNumberFacetsReturned(resp, ROLE_GROUP, 0);
+    }
+
+    @Test
+    public void limitedPatronAccessTest() throws Exception {
+        accessGroups = new AccessGroupSet(PUBLIC_PRINC);
+
+        SearchState searchState = new SearchState();
+        searchState.setFacetsToRetrieve(FACETS_TO_RETRIEVE);
+
+        SearchRequest request = new SearchRequest(searchState, accessGroups);
+        request.setApplyCutoffs(false);
+        SearchResultResponse resp = service.getFacetListResult(request);
+
+        assertNumberFacetsReturned(resp, PARENT_COLLECTION, 2);
+        assertFacetValueCount(resp, PARENT_COLLECTION, testCorpus.coll1Pid.getId(), 6);
+        assertFacetValueCount(resp, PARENT_COLLECTION, testCorpus.coll2Pid.getId(), 2);
+
+        assertNumberFacetsReturned(resp, CONTENT_TYPE, 2);
+        assertFacetValueCount(resp, CONTENT_TYPE, "^text", 3);
+        assertFacetValueCount(resp, CONTENT_TYPE, "^image", 1);
+
+        assertNumberFacetsReturned(resp, ROLE_GROUP, 3);
+        assertFacetValueCount(resp, ROLE_GROUP, "canViewOriginals|everyone", 12);
+        assertFacetValueCount(resp, ROLE_GROUP, "unitOwner|unitOwner", 11);
+        assertFacetValueCount(resp, ROLE_GROUP, "canManage|manager", 7);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void noAccessTest() throws Exception {
+        accessGroups = new AccessGroupSet(PUBLIC_PRINC);
+
+        SearchState searchState = new SearchState();
+        searchState.setFacetsToRetrieve(FACETS_TO_RETRIEVE);
+
+        SearchRequest request = new SearchRequest(searchState, accessGroups);
+        request.setRootPid(testCorpus.privateFolderPid);
+        request.setApplyCutoffs(false);
+        service.getFacetListResult(request);
+    }
+
+    @Test
+    public void filterByFacetNotBeingRetrievedTest() throws Exception {
```

Comment:
What's the difference between this and singleCollectionFacetTest on line 137?

Reply:
This test is verifying that you can filter by a facet which is not in the list of facets being retrieved. It is filtering on PARENT_COLLECTION, which is not in the facetsToRetrieve list, whereas it is in that list in `singleCollectionFacetTest`. I believe this situation didn't quite behave correctly initially.